File: shoring/flywheel/src/main/java/com/panopset/flywheel/TemplateLine.java
package com.panopset.flywheel;

public class TemplateLine {
  private final String line;
  private final Integer templateLineNumber;
  private final Integer templateCharIndex;
  
  public TemplateLine(String line, int templateCharIndex, int templateLineNumber) {
    this.line = line;
    this.templateCharIndex = templateCharIndex;
    this.templateLineNumber = templateLineNumber;
  }

  public TemplateLine(String line) {
    this(line, 0, 0);
  }

  public String getLine() {
    return line;
  }

  public Integer getTemplateLineNumber() {
    return templateLineNumber;
  }

  public Integer getTemplateCharIndex() {
    return templateCharIndex;
  }
  
  @Override
  public String toString() {
    return String.format(" line#: %5d, char#: 5%d line: %s", getTemplateLineNumber(), getTemplateCharIndex(), getLine()); 
  }
}

File: code/shoring/flywheel/src/main/kotlin/com/panopset/flywheel/TemplateLine.kt
package com.panopset.flywheel

class TemplateLine(
    val line: String,
    val templateCharIndex: Int, val templateLineNumber: Int) {
    constructor(line: String) : this(line, 0, 0)
    override fun toString(): String {
        return String.format(" line#: %5d, char#: 5%d line: %s", templateLineNumber, templateCharIndex, line)
    }
}
